src/mon/Elector.cc

```diff
@@ -36,13 +36,32 @@ Elector::Elector(Monitor *m) : logic(this,
 				     &peer_tracker, m->cct),
 			       peer_tracker(this, 12*60*60), // TODO: make configurable
 			       ping_timeout(2),
-			       mon(m), elector(this) {}
+			       mon(m), elector(this) {
+  bufferlist bl;
+  mon->store->get(Monitor::MONITOR_NAME, "connectivity_scores", bl);
+  if (bl.length()) {
+    bufferlist::const_iterator bi = bl.begin();
+    peer_tracker.decode(bi);
```

Comment:
Does anything bad happen if we persist really bad scores for a peer mon, go down for a long time, and then come back up with (old) bad scores when that peer has since been well behaved?  Does it still converge quickly?

Reply:
If we have bad scores for a mon it might change the leader, but if our bad score is the margin of victory I don't think there's any way it will matter (we have N alive monitors out of M total; the maximum score is N and a local score only contributes 0-1). The local scores will converge to the right one anyway.